lib/media/media_source_engine.js

```diff
@@ -915,12 +915,16 @@ shaka.media.MediaSourceEngine = class {
       }
     } else if (!mimeType.includes('/mp4') && !mimeType.includes('/webm') &&
         shaka.util.TsParser.probe(uint8ArrayData)) {
+      let parser;
       if (!this.tsParsers_.has(contentType)) {
-        this.tsParsers_.set(contentType, new shaka.util.TsParser());
+        parser = new shaka.util.TsParser();
+        this.tsParsers_.set(contentType, parser);
       } else {
-        this.tsParsers_.get(contentType).clearData();
+        parser = this.tsParsers_.get(contentType);
+        parser.clearData();
       }
-      const tsParser = this.tsParsers_.get(contentType).parse(uint8ArrayData);
+      parser.setDiscontinuitySequence(reference.discontinuitySequence);
+      const tsParser = parser.parse(uint8ArrayData);
       const startTime = tsParser.getStartTime(contentType);
```

Comment:
This code can be simplified a lot. In particular `const tsParser = parser.parse();` does not make a lot of sense, as `parse()` returns reference to itself.

```js
if (!this.tsParsers_.has(contentType)) {
  this.tsParsers_.set(contentType, new shaka.util.TsParser());
}
const parser = this.tsParsers_.get(contentType);
parser.clearData();
parser.setDiscontinuitySequence(reference.discontinuitySequence);
parser.parse(uint8ArrayData);
const startTime = parser.getStartTime(contentType);
```